### PR TITLE
Implement redis command 'BITOP' and 'BITFIELD'.

### PR DESCRIPTION
**Pull request by zyy17**
1. Add redis command `bitop`;
2. Add redis command `bitfield`;
3. Add new resp API `Array()` to output array that combines with multiple resp types.

### CODE REVIEW

File: encoding/resp/resp.go [hunk 1]
```diff
@@ -11,6 +11,51 @@ var (
...
+const (
+	SimpleString Type = '+'
+	Error        Type = '-'
+	Integer      Type = ':'
+	BulkString   Type = '$'
+	Array        Type = '*'
+)
+
+// The 'Value' design is referenced from https://github.com/tidwall/resp
+type Value struct {
```
Comment (shafreeck) [inline encoding/resp/resp.go]: Is the design of `Value` really necessary, it is a conflict style with the original design. Maybe we should think again and supply more reasons to change the design.

Comment (GhostComputing) [reply encoding/resp/resp.go]: I will change the PR to use the original way to construct response.

File: encoding/resp/resp.go [hunk 2]
```diff
@@ -11,6 +11,51 @@ var (
 	ErrInvalidProtocol = errors.New("invalid protocol")
 )
 
+type Type byte
+
+const (
+	SimpleString Type = '+'
+	Error        Type = '-'
```
Comment (YIDWang) [inline encoding/resp/resp.go]: This feature is currently supported in the package. Is this necessary？

**Comment by shafreeck** [comment]
ping @GhostComputing

**Pull request closed without merging**